### ISSUE TITLE
#11 Creating Custom Shapes not working

### ISSUE THREAD

**Issue by sureskumar** [turbobabr/Sketch-Plugins-Cookbook#13]
Hey, 11. Creating Custom Shapes does not seem to work in the latest Sketch v50. This is the error I am getting. Any help will be super appreciated. Cheers!

`-[NSBezierPath transformedPathUsingAffineTransform:]: unrecognized selector sent to instance 0x600000321a40`

**Comment by MareoRaft** [comment]
I get the same error.  It is caused by the code

    MSShapeGroup.shapeWithBezierPath(path)

Perhaps this is useful?: https://sketchplugins.com/d/858-getting-at-bezier-path-data

**Comment by sureskumar** [comment]
Thanks Matthew!

**Comment by MareoRaft** [comment]
@turbobabr 

So the solution is to replace the line

    var shape = MSShapeGroup.shapeWithBezierPath(path);

with the lines

    var mspath = MSPath.pathWithBezierPath(path);
    var shape = MSShapeGroup.shapeWithBezierPath(mspath);

.  I have tested this and it fixes functions 20 and 11 in the cookbook for Sketch version 50.2.

I am not sure why the issue was closed since it is not yet fixed in this repo.  I was thinking that this cookbook repo could be improved by adding an "archive" folder where you have a copy of the cookbook for each previous version of sketch.  Anyways, just a thought.  Have a great day!
